pjmedia/src/pjmedia-codec/and_vid_mediacodec.cpp

```diff
@@ -0,0 +1,1923 @@
+static pj_status_t encode_more_vpx(and_media_codec_data *and_media_data,
+                                   unsigned out_size,
+                                   pjmedia_frame *output,
+                                   pj_bool_t *has_more);
+static pj_status_t decode_vpx(pjmedia_vid_codec *codec,
+                              pj_size_t count,
+                              pjmedia_frame packets[],
+                              unsigned out_size,
+                              pjmedia_frame *output);
+#endif
+
+
+#if PJMEDIA_HAS_AND_MEDIA_H264
+static pj_str_t H264_sw_encoder[] = {{(char *)"OMX.google.h264.encoder\0",
+				      23}};
+static pj_str_t H264_hw_encoder[] =
+				  {{(char *)"OMX.qcom.video.encoder.avc\0", 26},
+			          {(char *)"OMX.Exynos.avc.Encoder\0", 22}};
+static pj_str_t H264_sw_decoder[] = {{(char *)"OMX.google.h264.decoder\0",
+				      23}};
+static pj_str_t H264_hw_decoder[] =
+				  {{(char *)"OMX.qcom.video.decoder.avc\0", 26},
+			          {(char *)"OMX.Exynos.avc.dec\0", 18}};
+#endif
+
+#if PJMEDIA_HAS_AND_MEDIA_VP8
+static pj_str_t VP8_sw_encoder[] = {{(char *)"OMX.google.vp8.encoder\0", 23}};
+static pj_str_t VP8_hw_encoder[] =
+				 {{(char *)"OMX.qcom.video.encoder.vp8\0", 26},
+			         {(char *)"OMX.Exynos.vp8.Encoder\0", 22}};
+static pj_str_t VP8_sw_decoder[] = {{(char *)"OMX.google.vp8.decoder\0", 23}};
+static pj_str_t VP8_hw_decoder[] =
+				 {{(char *)"OMX.qcom.video.decoder.vp8\0", 26},
+			         {(char *)"OMX.Exynos.vp8.dec\0", 18}};
+#endif
+
+#if PJMEDIA_HAS_AND_MEDIA_VP9
+static pj_str_t VP9_sw_encoder[] = {{(char *)"OMX.google.vp9.encoder\0", 23}};
+static pj_str_t VP9_hw_encoder[] =
+				 {{(char *)"OMX.qcom.video.encoder.vp9\0", 26},
+			         {(char *)"OMX.Exynos.vp9.Encoder\0", 22}};
+static pj_str_t VP9_sw_decoder[] = {{(char *)"OMX.google.vp9.decoder\0", 23}};
+static pj_str_t VP9_hw_decoder[] =
+				 {{(char *)"OMX.qcom.video.decoder.vp9\0", 26},
+			         {(char *)"OMX.Exynos.vp9.dec\0", 18}};
+#endif
+
+static struct and_media_codec {
+    int		       enabled;		  /* Is this codec enabled?	     */
+    const char	      *name;		  /* Codec name.		     */
+    const char        *description;       /* Codec description.              */
+    const char        *mime_type;         /* Mime type.                      */
+    pj_str_t          *encoder_name;      /* Encoder name.                   */
+    pj_str_t          *decoder_name;      /* Decoder name.                   */
+    pj_uint8_t	       pt;		  /* Payload type.		     */
+    pjmedia_format_id  fmt_id;		  /* Format id.   		     */
+    pj_uint8_t         keyframe_interval; /* Keyframe interval.              */
+
+    open_cb            open_codec;
+    process_encode_cb  process_encode;
+    encode_more_cb     encode_more;
+    decode_cb          decode;
+
+    pjmedia_codec_fmtp dec_fmtp;	  /* Decoder's fmtp params.	     */
+}
+and_media_codec[] {
```

Comment:
I got a compilation error here `definition of variable with array type needs an explicit size or an initializer and_media_codec[] {`. Not sure if it because of a compiler.
```suggestion
and_media_codec[] = {
```